### ISSUE TITLE
45.3 ATB Campaign wont load

### ISSUE THREAD

**Issue by Arakash0** [MegaMek/mekhq#1123]
### Environment
MeqHQ 0.45.3
Java version 1.8.0_181
Windows 7

### Description
Cant load several of the campaign saves i made in the last few hours. Ive attached the latest one that isn't working below.
I thought this might be this bug: https://github.com/MegaMek/mekhq/issues/1096
but the error message in MekHQlog seems to be different to the one listed there.

~~If there is any kind of workaround i could use to keep playing until the next release comes out i would be eternally grateful.~~
Edit: When sixlettervariables mentioned the refits might be the cause, found a workaround by deleting the refits in the save which let me load the campaign

### Files
[Campaign Save.zip](https://github.com/MegaMek/mekhq/files/2903750/Campaign.Save.zip)
[mechfiles.zip](https://github.com/MegaMek/mekhq/files/2903751/mechfiles.zip)
[mekhqlog.zip](https://github.com/MegaMek/mekhq/files/2903752/mekhqlog.zip)

**Comment by sixlettervariables** [comment]
Exception from log:
```
java.lang.ClassCastException: mekhq.campaign.parts.MekLocation cannot be cast to mekhq.campaign.parts.Armor
	at mekhq.campaign.io.CampaignXmlParser.parse(CampaignXmlParser.java:595)
	at mekhq.campaign.CampaignFactory.createCampaign(CampaignFactory.java:88)
	at mekhq.gui.dialog.DataLoadingDialog$Task.doInBackground(DataLoadingDialog.java:194)
	at mekhq.gui.dialog.DataLoadingDialog$Task.doInBackground(DataLoadingDialog.java:118)
	at javax.swing.SwingWorker$1.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at javax.swing.SwingWorker.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
```

**Comment by sixlettervariables** [comment]
Were you doing a refit?

https://github.com/MegaMek/mekhq/blob/83862b6d8caddfc3651f7c8d6411ae5b5147132c/MekHQ/src/mekhq/campaign/io/CampaignXmlParser.java#L595-L596

**Comment by Arakash0** [comment]
Yes.
From memory several refits were ongoing in the last few hours i was playing.

**Comment by sixlettervariables** [comment]
I believe what is happening is the armor supplies part's ID changes during the course of the refit, but is not being tracked. Still tracking down the root cause.

**Comment by sixlettervariables** [comment]
I'm tentatively closing this as fixed in #1216. This will land in either 0.45.5 or 0.46.
